pkg/vsphere/disk/disk.go

```diff
@@ -91,13 +106,22 @@ func (d *VirtualDisk) unlock() {
 
 func (d *VirtualDisk) setAttached(devicePath string) error {
 	if d.Attached() {
-		return fmt.Errorf("%s is already attached (%s)", d.DatastoreURI, devicePath)
```

Comment:
Not that your changes is wrong but maybe following could be little bit easier to read (aka bump to count when we don't return an error)

```
defer func() {
if err == nil {
    d.attachedRefs++
    log.Debugf("Increased attach references for %s to %d", d.DatastoreURI, d.attachedRefs)
}
```